### PR TITLE
core: Add "one-of" validation for number of kwargs

### PR DESCRIPTION
**Pull request by adamcik**

### CODE REVIEW

**Comment by adamcik** [comment]
Btw, I would be happy to package up the sum pattern in some `validation` helper. Just couldn't thing of a good name or to do it with min/max and raising or just returning the sum.

File: mopidy/core/library.py [hunk 1]
```diff
@@ -189,14 +189,9 @@ def lookup(self, uri=None, uris=None):
         .. deprecated:: 1.0
             The ``uri`` argument. Use ``uris`` instead.
         """
-        none_set = uri is None and uris is None
-        both_set = uri is not None and uris is not None
-
-        if none_set or both_set:
+        if sum(o is not None for o in [uri, uris]) != 1:
             raise ValueError("One of 'uri' or 'uris' must be set")
```
Comment (jodal) [inline mopidy/core/library.py]: Switch to double quotes around the values?

Change to "Exactly one of ..." for consistency and clearity?

File: mopidy/core/tracklist.py [hunk 2]
```diff
@@ -406,10 +406,9 @@ def add(self, tracks=None, at_position=None, uri=None, uris=None):
         .. deprecated:: 1.0
             The ``tracks`` and ``uri`` arguments. Use ``uris``.
         """
-        assert tracks is not None or uri is not None or uris is not None, \
-            'tracks, uri or uris must be provided'
-        # TODO: check that only one of tracks uri and uris is set...
-        # TODO: can at_position be negative?
+        if sum(o is not None for o in [tracks, uri, uris]) != 1:
+            raise ValueError(
+                'Exactly one of tracks, uri or uris must be provided')
```
Comment (jodal) [inline mopidy/core/tracklist.py]: Add double quotes around the kwarg names?

**Comment by jodal** [comment]
+1 with comments

**Comment by adamcik** [comment]
Done.

**Pull request merged**